File: ScalaSpark/src/main/scala/Study/ScalaSpark/WordCount.scala
package Study.ScalaSpark

import org.apache.spark.{SparkConf,SparkContext}
import org.apache.spark.rdd.{RDD}
 
import org.apache.spark.SparkConf
import org.apache.spark.SparkContext

object WordCount {
  
  val inputPath:String=""
  val outputPath:String=""
  
   def main(args:Array[String]):Unit={
      val conf = new SparkConf().setAppName("WordCount").setMaster("local")
      val sc =new SparkContext(conf)
      val lines: RDD[String] =sc.textFile(inputPath) 
      val words:RDD[String]=lines.flatMap(_.split(" "))
      val wordAndOne :RDD[(String, Int)]=words.map((_,1))
      val reduced : RDD[(String, Int)]=wordAndOne.reduceByKey(_+_)
      val sorted: RDD[(String, Int)] =reduced.sortBy(_._2, false)
      sorted.saveAsTextFile(outputPath)
      sc.stop()
   }
}

File: ScalaSpark/src/main/java/Spark/WordCount.java
package Spark;

import java.util.Arrays;

import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaPairRDD;
import org.apache.spark.api.java.JavaRDD;
import org.apache.spark.api.java.JavaSparkContext;

import scala.Tuple2;

public class WordCount {

	public static final String master  = "local[*]";
	public static final String basePath  ="hdfs://ns/BigData";
	public static final String inputPath  =basePath +"/WordCount.dat";
	public static final String outputPath =basePath +"/WordCount";


	public static void main(String[] args) {

		SparkConf conf = new SparkConf().setAppName("WordCount").setMaster(master);
		JavaSparkContext jsc = new JavaSparkContext(conf);
		JavaRDD<String> lines = jsc.textFile(inputPath);
		JavaRDD<String> words = lines.flatMap(line -> Arrays.asList(line.split(" ")).iterator());
		JavaPairRDD<String, Integer> wordAndOne=words.mapToPair(w->new Tuple2<>(w,1));
		JavaPairRDD<String, Integer> reduced=wordAndOne.reduceByKey((m,n)->m+n);
		JavaPairRDD<Integer, String>   swaped=reduced.mapToPair(f->f.swap());
		JavaPairRDD<Integer, String> sorted =swaped.sortByKey(false);
		JavaPairRDD<String, Integer> result=sorted.mapToPair(f->f.swap());
		result.saveAsTextFile(outputPath);
		jsc.stop();
		jsc.close();
	}

}
